### ISSUE TITLE
Double string on one link

### ISSUE THREAD

**Issue by valplusplusle** [eneko89/cachekill#2]
after usage, one image got a wrong replacing.

in the folder the image ist called: 
placeholder-valentin-f956d81ec00f7b2e5592ade6fe9f2983.webp

in the html file it was replaced with: 
placeholder-valentin-f956d81ec00f7b2e5592ade6fe9f2983-ef63272fee16e03b4c225a575f56864d.webp

**Comment by eneko89** [comment]
it looks like somehow, that file is getting hashed twice.
Could you add more context to this? Or even a test case to reproduce it?
it would be helpful and much appreciated 🙂

**Comment by eneko89** [comment]
Is this still a problem @valplusplusle ?
If so, could you please add more details so I can take a look at it?

**Comment by valplusplusle** [comment]
Hey Hey sorry i have holidays right now :D I am back on my working pc on monday and will take a look to share a full testable szenario:) Maybe i find a fix my self and make a merge request if i found something:)

**Comment by eneko89** [comment]
Great, thanks a lot!

**Comment by valplusplusle** [comment]
I have no clue why this is happening, but it is something with the images and nothing in the code.
I can only reproduce it with one specific image i attache it into the issue.
I use webp as format for every picture since google loves it. This is the only one with this bug, doesnt matter if i rename it. If i just use another one everything works fine.... so i think we can close this strange issue :D 
(but i found out, that strange file names/characters can also break it)

[images.zip](https://github.com/eneko89/cachekill/files/6377129/images.zip)

**Comment by eneko89** [comment]
Do you have the code in a public repo? I could take a look, cause it's really unlikely, I'd say impossible, for this issue to be related to a particular image. The content of the file is only used to compute the hash, renaming and replacing logic is not related to whatever the file contains.

It's more likely related to the build process or to some bug in cachekill...

**Comment by valplusplusle** [comment]
There is a lot of sensible data in it... i will take a look on monday if there is a way to show the full original scenario

**Comment by valplusplusle** [comment]
After building up a version which i can upload for testing, i figured it out. It was just because the the number of times using the tool was not in sync. 
The problem is, if you want to use it multiple times, you always have to delete the old cachekilled names and paths. The problem was: One image was already cachekilled. everytime i use cachekill it obviousley chachekill the cachekilled ones.
Fist Time: placeholder-matze.webp -> placeholder-matze-ea1b8bf99d354661ae392149302cd20b.webp
Second Time : placeholder-matze-ea1b8bf99d354661ae392149302cd20b -> placeholder-matze-ea1b8bf99d354661ae392149302cd20b-ea1b8bf99d354661ae392149302cd20b.webp

And so on. Problem was: one picture goes in a page which was 5 time going through cachekill so at the next time it broke the path of this specific image.

Long story short -> i dont use cachekill this way it was just for first testing. I dont cachekill a already cachekilled page i want to use it on deployment.

So there is no issue here, nothing to see ! :D Thanks for all.

But i nice feature would be not to copy the cachekilled stuff in the same path. Just create a new folder where only the images with the cachekill fingerprint are inside. So its a little bit a struggle to deploy because you always have to remove the old pictures.

**Comment by eneko89** [comment]
Gotcha!

I will consider adding an option to specify a destination folder. What I personally do is to copy files to another folder (e.g. from `assets` to `dist`) as part of the build process, and then rename them (`-r` option) instead of copying. Take a look at t[he build procees of my webpage](https://github.com/eneko89/eneko.me/blob/1c89c9f396f374b5cefe84b7133c78a341aa45cb/package.json#L15) if you want.

Hope it helps 🙂
